uber/models/department.py

```diff
@@ -0,0 +1,271 @@
+from datetime import timedelta
+
+from sideboard.lib import cached_property
+from sideboard.lib.sa import CoerceUTF8 as UnicodeText, \
+    UTCDateTime, UUID
+from sqlalchemy import select
+from sqlalchemy.orm import column_property
+from sqlalchemy.schema import ForeignKey, Table, UniqueConstraint
+from sqlalchemy.types import Boolean, Float, Integer
+
+from uber.config import c
+from uber.models import MagModel
+from uber.models.attendee import Attendee
+from uber.models.types import default_relationship as relationship, \
+    Choice, DefaultColumn as Column
+
+
+__all__ = [
+    'department_membership_job_role', 'department_membership_request',
+    'job_required_job_role', 'Department', 'DepartmentMembership', 'Job',
+    'JobRole', 'Shift']
+
+
+department_membership_job_role = Table(
```

Comment:
Hmmm.. I know this is how the SQLAlchemy docs do these join tables (iirc), but it's kinda weird to use declarative style for some tables and... um, whatever this is for other tables. Is there any particular reason to do it this way?

Reply:
This particular table doesn't have any additional properties. It's sole purpose is to join the `department_membership` and `job_role` tables. We should never need to access the rows in this table directly, so we don't really need a model class mapped to it.

That doesn't mean we _can't_ map a model class to it, but we wouldn't gain anything by doing it. There are some potential downsides to mapping a model class onto an association table. If we make our mapped model class a subclass of `MagModel` like we do all our other classes, it will _also_ have an `id` primary key column. Just having an additional column creates some minor memory and storage overhead (not enough to really matter). The real drawback is that because the `id` column is a primary key, it is necessarily indexed, which creates some overhead when inserting new rows. Again, it's not enough to really make an impact, but if there's no reason to map a class to the table we're better off avoiding it.

Reply:
That totally makes sense!

Since this style of doing tables is new to the codebase, do you mind leaving a short comment explaining that they're just for many-to-many joins and they're left this way to improve performance?